editor/src/components/editor/actions/actions.ts

```diff
@@ -1789,18 +1790,26 @@ export const UPDATE_FNS = {
       )
     }
   },
-  PUSH_TOAST: (action: PushToast, editor: EditorModel, dispatch: EditorDispatch): EditorModel => {
-    setTimeout(() => dispatch([popToast()], 'everyone'), 5500)
+  ADD_TOAST: (action: AddToast, editor: EditorModel, dispatch: EditorDispatch): EditorModel => {
+    const uniqueToastId = UUID()
+
+    if (!action.toast.persistent) {
+      setTimeout(() => dispatch([removeToast(uniqueToastId)], 'everyone'), 5500)
+    }
+    const newToast: Notice = {
+      ...action.toast,
+      id: uniqueToastId,
```

Comment:
See comment above; here I'm basically nuking whatever ID has been set, but that then requires two types (or alternatively not enforcing uniqueness, unless there's a type _for that_?)

Reply:
If we make ID non-optional then we can create the UUID in the function that creates `AddToast` actions, removing the need for this new object